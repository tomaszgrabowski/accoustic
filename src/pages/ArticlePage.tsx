import React, { useEffect, useState } from 'react';
import { stripHtmlTags } from '../functions/functions';
import { BASE_URL } from '../constants';
import { ArticleContext } from '../contexts/ArticleContext';
import { IApiResponse } from '../models/IApiResponse';
import { getArticleByGuid } from '../services/data-service';

const ArticlePage = () => {
    const [article, setArticle] = useState<IApiResponse | undefined>( undefined );
    const [error, setError] = useState<string>();
    
    useEffect( () => {
        getArticleByGuid()
            .then( ( response: IApiResponse ) => {
                setArticle( response );
            } )
            .catch( err => {
                setError( err );
            } );
    }, [] );
    
    return (
        <ArticleContext.Provider value={ article }>
            { article &&
            <article>
              <header>{ article.elements.heading.value }</header>
              <time>{ article.elements.date.value }</time>
              <main>{ article.elements.body.values.map( value => stripHtmlTags( value ) ) }</main>
              <footer>{ article.elements.author.value }</footer>
              <aside>
                <figure>
                  <img src={ `${ BASE_URL }${ article.elements.mainImage.value.leadImage.url }` }
                       alt={ article.elements.mainImage.value.leadImageCaption.value }/>
                  <figcaption>{ article.elements.mainImage.value.leadImageCaption.value }</figcaption>
                </figure>
              </aside>
            </article>
            }
        </ArticleContext.Provider>
    );
};

export default ArticlePage;
