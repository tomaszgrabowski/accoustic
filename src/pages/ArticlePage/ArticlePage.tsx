import React, { useEffect, useState } from 'react';
import Article from '../../components/Article/Article';
import { ArticleContext } from '../../contexts/ArticleContext';
import { IApiResponse } from '../../models/IApiResponse';
import { getArticleByGuid } from '../../services/data-service';

const ArticlePage = () => {
    const [article, setArticle] = useState<IApiResponse>();
    const [loading, setLoading] = useState<boolean>( false );
    const [error, setError] = useState<boolean>( false );
    
    useEffect( () => {
        setLoading( true );
        getArticleByGuid()
            .then( ( response: IApiResponse ) => {
                setArticle( response );
                setLoading( false );
            } )
            .catch( err => {
                setError( err );
                setLoading( false );
            } );
    }, [] );
    
    return (
        <ArticleContext.Provider value={ article }>
            <Article loading={ loading } article={ article } error={ error }/>
        </ArticleContext.Provider>
    );
};

export default ArticlePage;
