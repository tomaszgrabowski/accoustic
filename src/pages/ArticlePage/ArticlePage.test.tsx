import React from 'react';
import renderer from 'react-test-renderer';
import { apiResponse } from '../../test/mock';
import ArticlePage from './ArticlePage';

describe( 'ArticlePage', () => {
    test( 'given data when render then show section', () => {
        jest.spyOn( React, 'useReducer' ).mockReturnValue( [
            { error: false, loading: false, article: apiResponse },
            () => {
            }
        ] );
        const wrapper = renderer.create( <ArticlePage/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
