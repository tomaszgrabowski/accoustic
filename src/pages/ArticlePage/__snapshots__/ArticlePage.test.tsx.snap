// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage given data when render then show section 1`] = `
<article>
  <header
    className="article-header"
  >
    <h2>
      Staffordshire Terrier Pups
    </h2>
  </header>
  <div>
    <main>
      <section
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry&#39;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
",
          }
        }
      />
      <section
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry&#39;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
",
          }
        }
      />
      <section
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry&#39;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
",
          }
        }
      />
      <section
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry&#39;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
",
          }
        }
      />
    </main>
    <aside>
      <figure>
        <img
          alt="Staffordshire Terrier Pups"
          src="https://my12.content-cms.com/859f2008-a40a-4b92-afd0-24bb44d10124/dxresources/9349/93493572-ea5b-4c27-89cf-8a5c3ee617ad.jpg"
        />
        <figcaption>
          Staffordshire Terrier Pups
        </figcaption>
      </figure>
    </aside>
  </div>
  <footer>
    <span>
      Gabe Bravo
    </span>
    <small>
      13-00-2019 18:00
    </small>
  </footer>
</article>
`;
